core/lib/eth_client/src/clients/http/query.rs

```diff
@@ -103,11 +111,22 @@ where
                 .with_arg("chunk_size", &chunk_size)
                 .with_arg("block", &chunk_end)
                 .await?;
-            history.extend(fee_history.base_fee_per_gas);
+
+            for (base, blob) in fee_history
+                .base_fee_per_gas
+                .into_iter()
+                .zip(fee_history.base_fee_per_blob_gas)
```

Comment:
Dumb question: Do all Web3 providers support these new fields? If they are not returned, this field here will be empty, so that `history` will not be filled either, even if `fee_history.base_fee_per_gas` is filled.

Reply:
I've checked Infura, Alchemy and `reth` -- all seem to support it.

Reply:
Actually, it seems that it's [a part of specification now](https://ethereum.github.io/execution-apis/api-documentation/).

Reply:
Got it. IMO, it'd still be safer to fill these value with zeros at least if `base_fee_per_blob_gas` is empty (don't know what to do if it's non-empty, but shorter than `base_fee_per_gas`; maybe return an error). OTOH, this may be an unnecessary complication, so feel free to ignore.

Reply:
I've added a `tracing::error` for cases where the lengths of returned arrays are not equal.
Filling with zeros doesn't sound right -- we rely on these values in production, and using zeros silently will cause blob transactions to stuck probably.
Returning an error doesn't feel right either -- if occurs, it will probably either stop or crash many components.

That said, I've checked a few more providers and they do seem to support it; moreover they specify that the values returned must be 0 for pre EIP-4844 blocks (I've also adapted our API to match this behavior).